server/src/test/java/org/elasticsearch/repositories/blobstore/BlobStoreRepositoryTests.java

```diff
@@ -137,7 +140,7 @@ public void testRetrieveSnapshots() throws Exception {
 
     public void testReadAndWriteSnapshotsThroughIndexFile() throws Exception {
         final BlobStoreRepository repository = setupRepo();
-
+        final long pendingGeneration = getPendingGeneration(repository);
```

Comment:
The generation here will now be something other than `-1` because we are using the same repository name across tests. I intentionally didn't add a repository cleanup step between tests here, since this serves as a neat test (and illustration) of how the pending generation is kept consistent no matter what happens to the repo contents.